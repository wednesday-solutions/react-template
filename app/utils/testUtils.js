import React from 'react';
import { IntlProvider } from 'react-intl';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { browserHistory, BrowserRouter, Router } from 'react-router-dom';
import { ThemeProvider } from '@mui/material/styles';
import configureStore from '@app/configureStore';
import { DEFAULT_LOCALE, translationMessages } from '@app/i18n';
import ConnectedLanguageProvider from '@containers/LanguageProvider';
import { IntlGlobalProvider } from '@components/IntlGlobalProvider';
import { theme } from '@containers/App';

export const renderWithIntl = (children) =>
  render(
    <IntlProvider locale={DEFAULT_LOCALE} messages={translationMessages[DEFAULT_LOCALE]}>
      <IntlGlobalProvider>{children}</IntlGlobalProvider>
    </IntlProvider>
  );

export const renderProvider = (children, history) => {
  const store = configureStore({}, browserHistory).store;
  return render(
    <Provider store={store}>
      <ConnectedLanguageProvider messages={translationMessages}>
        <ThemeProvider theme={theme}>
          {history ? <Router history={history}>{children}</Router> : <BrowserRouter>{children}</BrowserRouter>}
        </ThemeProvider>
      </ConnectedLanguageProvider>
    </Provider>
  );
};
export const timeout = (ms) => new Promise((resolve) => setTimeout(resolve, ms));
export const apiResponseGenerator = (ok, data) => ({
  ok,
  data
});

export const getStyles = () => {};
