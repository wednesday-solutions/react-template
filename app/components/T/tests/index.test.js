/**
 *
 * Tests for T
 *
 */

import React from 'react';
import { matchers } from '@emotion/jest';
import { renderWithIntl } from '@utils/testUtils';
import { T } from '../index';

expect.extend(matchers);

describe('<T /> component tests', () => {
  it('should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(<T />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should contain 1 T component', () => {
    const { getAllByTestId } = renderWithIntl(<T />);
    expect(getAllByTestId('t').length).toBe(1);
  });

  it('should contain render the text according to the id', () => {
    const { getAllByText } = renderWithIntl(<T id="repo_list" />);
    expect(getAllByText(/Repository List/).length).toBe(1);
  });

  it('should have a margin-bottom of 5px when we pass marginBottom as 5', () => {
    const props = {
      marginBottom: 5,
      id: 'repo_list'
    };
    const { getByTestId } = renderWithIntl(<T {...props} />);
    const styledText = getByTestId('t');
    expect(styledText).toHaveStyle({ marginBottom: '5px' });
  });
});
