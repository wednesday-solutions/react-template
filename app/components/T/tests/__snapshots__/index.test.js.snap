// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<T /> component tests should render and match the snapshot 1`] = `
<body>
  <div>
    <p
      class="T__StyledText-gjlic1-0 cbAvsY"
      data-testid="t"
    />
  </div>
</body>
`;
